Source/JavaScriptCore/jit/RegisterSet.cpp

```diff
@@ -421,6 +421,22 @@ RegisterSet RegisterSetBuilder::allScalarRegisters()
     return result;
 }
 
+#if ENABLE(WEBASSEMBLY)
+RegisterSet RegisterSetBuilder::wasmPinnedRegisters(MemoryMode memoryMode)
+{
+    RegisterSet result;
+    if constexpr (GPRInfo::wasmBaseMemoryPointer != InvalidGPRReg)
+        result.add(GPRInfo::wasmBaseMemoryPointer, IgnoreVectors);
+    if constexpr (GPRInfo::wasmContextInstancePointer != InvalidGPRReg)
+        result.add(GPRInfo::wasmContextInstancePointer, IgnoreVectors);
+    if (memoryMode == MemoryMode::BoundsChecking) {
+        if constexpr (GPRInfo::wasmBoundsCheckingSizeRegister != InvalidGPRReg)
```

Comment:
Should we check the `constexpr` expression first before the `memoryMode` test?

Reply:
Changed.